public/app/percona/shared/core/reducers.ts

```diff
@@ -250,14 +252,61 @@ export const deleteKubernetesAction = createAsyncThunk(
 
 export const addKubernetesAction = createAsyncThunk(
   'percona/addKubernetes',
-  async (args: { kubernetesToAdd: NewKubernetesCluster; token?: CancelToken }, thunkAPI): Promise<void> => {
+  async (
+    args: { kubernetesToAdd: NewKubernetesCluster; setPMMAddress?: boolean; token?: CancelToken },
+    thunkAPI
+  ): Promise<void> => {
+    if (args.setPMMAddress) {
+      await thunkAPI.dispatch(updateSettingsAction({ body: { pmm_public_address: window.location.host } }));
+      await new Promise((resolve) => setTimeout(resolve, SETTINGS_TIMEOUT));
+    }
     await withAppEvents(KubernetesService.addKubernetes(args.kubernetesToAdd, args.token), {
       successMessage: 'Cluster was successfully registered',
     });
     await thunkAPI.dispatch(fetchKubernetesAction());
   }
 );
 
+export const addDbClusterAction = createAsyncThunk(
```

Comment:
@fabio-silva can you explain why we need this service? I 'm trying to understand and the name looks like we should have something similar

Reply:
Is it becouse you changed step-progress and now we need "loading" props on addingCluster action?

Reply:
Yeah, I added this so that the loading comes from the state itself. We had too many loadings down there, the logic is easier if it comes from Redux